lib/net.js

```diff
@@ -1166,6 +1237,64 @@ function lookupAndConnect(self, options) {
   debug('connect: dns options', dnsopts);
   self._host = host;
   const lookup = options.lookup || dns.lookup;
+
+  if (dnsopts.family !== 4 && dnsopts.family !== 6 && options.autoDetectFamily) {
```

Comment:
I think `autoDetectFamily` should be `true` by default - this problem is hitting quite a few folks given the DNS change.